src/gh_changes_since.rs

```diff
@@ -0,0 +1,72 @@
+use std::sync::Arc;
+
+use anyhow::Context as _;
+use axum::{
+    extract::{Path, State},
+    response::{IntoResponse, Redirect, Response},
+};
+use hyper::StatusCode;
+
+use crate::{github, handlers::Context, utils::AppError};
+
+pub async fn gh_changes_since(
+    Path((owner, repo, pr_num, oldbasehead)): Path<(String, String, u64, String)>,
+    State(ctx): State<Arc<Context>>,
+) -> axum::response::Result<Response, AppError> {
+    let Some((oldbase, oldhead)) = oldbasehead.split_once("..") else {
+        return Ok((
+            StatusCode::BAD_REQUEST,
+            format!("`{oldbasehead}` is not in the form `base..head`"),
+        )
+            .into_response());
+    };
+
+    let repos = ctx
+        .team
+        .repos()
+        .await
+        .context("unable to retrieve team repos")?;
+
+    // Verify that the request org is part of the Rust project
+    let Some(repos) = repos.repos.get(&owner) else {
```

Comment:
We might want to extract this check, together with the `repos` load, it's in a bunch of handlers at this point.